pkgs/unified_analytics/test/unified_analytics_test.dart

```diff
@@ -814,26 +894,39 @@ $initialToolName=${ConfigHandler.dateStamp},$toolsMessageVersion
             'constants.dart need to match\n'
             'Pubspec: $version && constants.dart: $kPackageVersion\n\n'
             'Make sure both are the same');
+
+    // Parse the contents of the change log file
+    final String changeLogFirstLineString =
+        io.File('CHANGELOG.md').readAsLinesSync().first;
+    expect(changeLogFirstLineString.substring(3), kPackageVersion,
+        reason: 'The CHANGELOG.md file needs the first line to '
+            'be the same version as the pubspec and constants.dart');
   });
 
   test('Null values for flutter parameters is reflected properly in log file',
       () {
-    final Analytics secondAnalytics = Analytics.test(
-      tool: secondTool,
-      homeDirectory: home,
-      measurementId: 'measurementId',
-      apiSecret: 'apiSecret',
-      // flutterChannel: flutterChannel,           THIS NEEDS TO REMAIN REMOVED
-      // toolsMessageVersion: toolsMessageVersion, THIS NEEDS TO REMAIN REMOVED
-      toolsMessage: toolsMessage,
-      flutterVersion: 'Flutter 3.6.0-7.0.pre.47',
-      dartVersion: 'Dart 2.19.0',
-      fs: fs,
-      platform: platform,
-    );
+    // Use a for loop two initialize the second analytics instance
+    // twice to account for no events being sent on the first instance
+    // run for a given tool
+    Analytics? secondAnalytics;
+    for (int i = 0; i < 2; i++) {
+      secondAnalytics = Analytics.test(
+        tool: secondTool,
+        homeDirectory: home,
+        measurementId: 'measurementId',
+        apiSecret: 'apiSecret',
+        // flutterChannel: flutterChannel,  THIS NEEDS TO REMAIN REMOVED
+        toolsMessageVersion: toolsMessageVersion,
+        toolsMessage: toolsMessage,
+        flutterVersion: 'Flutter 3.6.0-7.0.pre.47',
+        dartVersion: 'Dart 2.19.0',
+        fs: fs,
+        platform: platform,
+      );
+    }
 
     // Send an event and check that the query stats reflects what is expected
-    secondAnalytics.sendEvent(
+    secondAnalytics!.sendEvent(
```

Comment:
This file is getting long, can you break it up into smaller files?

Reply:
Yep, getting around to it, captured in this issue https://github.com/dart-lang/tools/issues/10